compiler/luci/pass/src/CircleOptimizer.cpp

```diff
@@ -109,6 +109,11 @@ void CircleOptimizer::optimize(loco::Graph *g) const
   logo::Phase phase;
 
   /* TRANSFORM DECLARATION BEGIN */
+
+  // Shape and type inference is needed for doing below transformations
+  phase.emplace_back(std::make_unique<luci::ShapeInferencePass>());
+  phase.emplace_back(std::make_unique<luci::TypeInferencePass>());
```

Comment:
Some passes can change the type of nodes. Is it ok not to update the types after those passes run?

Reply:
> Some passes can change the type of nodes.

If so, the phase runner will run the whole passes again, so it will be ok. Never mind this comment :)